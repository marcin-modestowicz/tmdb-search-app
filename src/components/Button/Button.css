.root {
  display: inline-block;
  padding: 0 0.5em;
  cursor: pointer;
  border-radius: 0.25em;
  font-family: sans-serif;
  background-color: cornflowerBlue;
  color: white;
  font-size: 1em;
  line-height: 2em;
}

.root:active {
  background-color: royalBlue;
}
