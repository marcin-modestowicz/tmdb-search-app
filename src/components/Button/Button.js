// @flow

import React, { Component } from "react";
import styles from "./Button.css";

/* global React$Element */

type Props = {
  children: React$Element<*>,
  onClick: () => void
};

const Button = ({ children, onClick }: Props) => (
  <div role="button" className={styles.root} onClick={onClick}>
    {children}
  </div>
);

export default Button;
