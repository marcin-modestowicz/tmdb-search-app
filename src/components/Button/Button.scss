.root {
  padding: 0 0.5em;
  cursor: pointer;
  border-radius: 0.25em;
  font-family: sans-serif;
  background-color: cornflowerBlue;
  border: 1px solid cornflowerBlue;
  color: white;
  font-size: 1em;
  line-height: 2em;
  text-align: center;

  &:active {
    background-color: royalBlue;
    border-color: royalBlue;
  }
}
