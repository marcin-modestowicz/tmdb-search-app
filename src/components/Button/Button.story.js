// @flow

import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import Button from "./Button";

storiesOf("Button", module).add("simple", () => (
  <Button onClick={action("clicked")}>Button</Button>
));
