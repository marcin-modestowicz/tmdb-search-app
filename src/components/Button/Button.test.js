// @flow
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Button from "./Button";

describe("Button component", () => {
  let button;
  let onClick;

  beforeEach(() => {
    onClick = jest.fn();
    button = shallow(<Button onClick={onClick}>Button</Button>);
  });

  test("should render", () => {
    expect(toJson(button)).toMatchSnapshot();
  });

  test("onClick handler should be triggered", () => {
    button.simulate("click");
    expect(onClick).toHaveBeenCalled();
  });
});
