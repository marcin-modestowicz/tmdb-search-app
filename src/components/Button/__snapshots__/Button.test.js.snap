// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render 1`] = `
<div
  className="root"
  onClick={[Function]}
  role="button"
>
  Button
</div>
`;
